src/Cache/PropertyResolver/AssociationResolver.php

```diff
@@ -73,7 +73,9 @@ public function resolveSubscription(
 
         if (null !== $if = $routeMetadata->purgeOn->if) {
             $expression = (string) $if;
-            $if = new Expression(str_replace('obj', 'obj.'.$this->createGetter($associationClass, $associationTarget), $expression));
+            $getter = $this->createGetter($associationClass, $associationTarget);
+            $inverseIf = str_replace('obj', 'obj.'.$this->createGetter($associationClass, $associationTarget), $expression);
```

Comment:
@Brajk19 Should we search for `obj.` instead?

Reply:
that wouldn't work if we have checks like `obj !== null` or custom EL functions: `foo(obj)` where there is no dot